src/storage/StorageServer.cpp

```diff
@@ -0,0 +1,166 @@
+    return true;
+}
+
+bool StorageServer::start() {
+    ioThreadPool_ = std::make_shared<folly::IOThreadPoolExecutor>(FLAGS_num_io_threads);
+    workers_ = apache::thrift::concurrency::PriorityThreadManager::newPriorityThreadManager(
+                                 FLAGS_num_worker_threads, true /*stats*/);
+    workers_->setNamePrefix("executor");
+    workers_->start();
+
+    std::string clusteridFile =
+        folly::stringPrintf("%s/%s", dataPaths_[0].c_str(), "/storage.cluster.id");
+    clusterMan_
+        = std::make_unique<nebula::meta::ClusterManager>("", clusteridFile);
+    if (!clusterMan_->loadClusterId()) {
+        LOG(INFO) << "storaged misses clusterId";
+    }
+    // Meta client
+    metaClient_ = std::make_unique<meta::MetaClient>(ioThreadPool_,
+                                                     metaAddrs_,
+                                                     localHost_,
+                                                     clusterMan_.get(),
+                                                     true);
+    if (!metaClient_->waitForMetadReady()) {
+        LOG(ERROR) << "waitForMetadReady error!";
+        return false;
+    }
+
+    gFlagsMan_ = std::make_unique<meta::ClientBasedGflagsManager>(metaClient_.get());
+    gFlagsMan_->init();
+
+    LOG(INFO) << "Init schema manager";
+    schemaMan_ = meta::SchemaManager::create();
+    schemaMan_->init(metaClient_.get());
+
+    LOG(INFO) << "Init kvstore";
+    kvstore_ = getStoreInstance();
+
+    if (nullptr == kvstore_) {
+        LOG(ERROR) << "Init kvstore failed";
+        return false;
+    }
+
+    if (!initWebService()) {
+        LOG(ERROR) << "Init webservice failed!";
+        return false;
+    }
+
+    auto handler = std::make_shared<StorageServiceHandler>(kvstore_.get(), schemaMan_.get());
+    try {
+        LOG(INFO) << "The storage deamon start on " << localHost_;
+        tfServer_ = std::make_unique<apache::thrift::ThriftServer>();
+        tfServer_->setPort(FLAGS_port);
+        tfServer_->setReusePort(FLAGS_reuse_port);
+        tfServer_->setIdleTimeout(std::chrono::seconds(0));  // No idle timeout on client connection
+        tfServer_->setIOThreadPool(ioThreadPool_);
+        tfServer_->setThreadManager(workers_);
+        tfServer_->setInterface(std::move(handler));
+        tfServer_->setStopWorkersOnStopListening(false);
+        tfServer_->serve();  // Will wait until the server shuts down
+    } catch (const std::exception& e) {
+        LOG(ERROR) << "Start thrift server failed, error:" << e.what();
+        return false;
+    }
+    return true;
+}
+
+void StorageServer::stop() {
+    if (stopped_) {
+        LOG(INFO) << "All services has been stopped";
+        return;
+    }
+    if (webStatus_ == Status::RUNNING) {
+        nebula::WebService::stop();
+        webStatus_ = Status::STOPPED;
+    }
+    if (metaClient_) {
+        metaClient_->stop();
+    }
+    if (kvstore_) {
+        kvstore_.reset();
```

Comment:
maybe here need some comment that why we can reset kvstore before tfServer that tfServer is dependence on kvstore.

Reply:
That's a good question!  And it is some hard to explain. 
In short,  after we stop ioThreadPool  in kvstore dtor,   it will cut-off the relationship with acceptors in storage thrift service.